assets/src/edit-story/app/story/effects/usePageBackgrounds.js

```diff
@@ -54,7 +58,8 @@ function usePageBackgrounds({ currentPage, setBackgroundElement, addElement }) {
     });
     addElement({ element });
     setBackgroundElement({ elementId: element.id });
-  }, [addElement, currentPage, setBackgroundElement]);
+    clearHistory();
```

Comment:
I really don't understand why history is cleared here. Could you clarify?

Reply:
Right above we have a `createNewElement` that we use to create a white rectangle for new stories. It is incrementing an unnecessary new history version, which also enables the prevent window unload. Adding this `clearHistory` call, we reset the history and ensure that starting new stories will not create a fake history change only because we add a fake (since its an element) white background on canvas.

Reply:
If we remove this, the user will always receive a prevent window unload prompt, if try to reload the page of a new story, even without direct changes.

Reply:
But this also means, that adding a page and pressing reload doesn't warn the user, even though you've clearly made a change.

It really only should be the very first page added to the story, that should be ignored for these purposes - not subsequent pages.

Also not that @miina is changing the logic of the page background in #1129 .

Reply:
Since we are adding an element (for new stories), we should expect that a new history state will be handled. Should I add an independent `new-page` listener to prevent window unload when the user handles the action to add a New Page and remove it as soon as the user Save the story? So we can stay with this clearHistory without problems.

Reply:
I'd rather prevent unload was slightly broken than workaround this issue this way. Let's remove `clearHistory` from here for now.

Reply:
No. I don't think this is the right place to solve this problem. I think the issue with the new-page will have to be solved separately. From the point of view of `usePreventWindowUnload` as a user of the history system, it should be reasonable to demand that the history is "clean".

Reply:
Exactly. Also, new stories without any user intentional changes can't create history state updates. That would fix the problem. Anyway, let me remove `clearHistory` for now. Thanks @dvoytenko and @barklund for the insights here.

Reply:
With @miina's latest solution in #1129 this might actually be fixed magically, as the [`restore` in `useLoadStory`](https://github.com/google/web-stories-wp/blob/master/assets/src/edit-story/app/story/effects/useLoadStory.js#L127-L133) will already have the background element present ([as it will be added in `createPage`](https://github.com/google/web-stories-wp/blob/master/assets/src/edit-story/app/story/effects/useLoadStory.js#L91)) and thus not cause a history change.

Do however make a note to check this (a new issue perhaps) once both these issues are merged. If it's still a problem then, we'll deal with it later.

Reply:
Just issued on #1137.

Reply:
Correct, #1129 will address that, too.